### PR TITLE
IGMGuesses and Qt5

### PR DESCRIPTION
**Pull request by profxj**
Quite a few edits to refactor for PyQt5.

Also had to handle the linetools refactor away from QTables.

igmguesses is working fine for me on Ubuntu and Python 3.6

### CODE REVIEW

**Comment by profxj** [comment]
@ntejos should review *after* PR #97 

@jessicawerk should be able to run smoothly again provided 
she changes over to PyQt5.

File: pyigm/guis/igmguesses.py
Comment (ntejos) [inline pyigm/guis/igmguesses.py:1453]: There is an "unwanted" behaviour with ``EditBox`` now. Basically, when I edit, say the b-value to ``10``, I need to write "1" and "0". Right after I write "1" in the box, the fit is updated to ``b=1`` km/s before letting me put the "0". This is somewhat annoying.

File: pyigm/guis/igmguesses.py
Comment (ntejos) [inline pyigm/guis/igmguesses.py]: May be related to this(these) new call(s).

File: pyigm/guis/igmguesses.py
Comment (ntejos) [inline pyigm/guis/igmguesses.py:1556]: Or may be related to this new way to ``update_component()``. Seems we need a way to tell the GUI that the user is done putting strings into the boxes before setting ``self.update = True``

File: pyigm/guis/igmguesses.py
Comment (ntejos) [reply pyigm/guis/igmguesses.py:1453]: Also, if I remove all the values, i.e. leaving only the string ``""``, the GUI breaks because cannot convert that string to a float. The same happens if I accidentally add a string that would make ``float(str)`` fail.
Update: I added a commit that deals with this, but the first issue remains (see previous comment).

File: pyigm/guis/igmguesses.py
Comment (ntejos) [inline pyigm/guis/igmguesses.py]: This bit here was added by me to address a previous issue; please review.

File: pyigm/guis/igmguesses.py
Comment (ntejos) [inline pyigm/guis/igmguesses.py:1736]: I also added this bit from a while ago (I had forgotten about this commit), but seems harmless here (only applies for LineList H2).

**Review by ntejos — APPROVED** [review]
This seems to be working, although some notes before merging:
- Check my comment about the annoying behaviour in update_component(), in case you see an easy fix.
- Check the couple of commits I added to solve a simple issue + an old one that deals with H2 LineList (unrelated to this PR, but harmless)

File: pyigm/guis/igmguesses.py
Comment (profxj) [reply pyigm/guis/igmguesses.py:1453]: Yeah, this is the new functionality of EditBox.  If you can figure
out how to stop it, we should change it.

File: pyigm/guis/igmguesses.py
Comment (profxj) [reply pyigm/guis/igmguesses.py]: These new calls are necessary in PyQT5.  I think

File: pyigm/guis/igmguesses.py
Comment (profxj) [reply pyigm/guis/igmguesses.py:1556]: Trust me, I fussed with this quite a bit.

Things were crashing a lot previously..

File: pyigm/guis/igmguesses.py
Comment (profxj) [reply pyigm/guis/igmguesses.py:1736]: ok

File: pyigm/guis/igmguesses.py
Comment (ntejos) [reply pyigm/guis/igmguesses.py:1453]: It seems to me that we would need to create a button on the widget like "Update", for when the user is done editing the box(es) the values are retrieved and updated after pressing such button. What do you think?

Comment (profxj) [reply pyigm/guis/igmguesses.py:1453]: Feel free to do so.  
Good luck..

**Comment by ntejos** [comment]
Here is an attempt at the "setzbNc" issue mentioned before. I have now added a button "Update" that only when pressed the values are updated from the values given by the user in the widget(s). This now includes the "comment" widget (because the annoying behaviour was also present in comment). 

TODO: Redefine the layout for the "update button", ideally has to go below all four "z", "N", "b", "comment" to make it implicit that the comment needs to be updated using the button too. When I tried to move the button around, I lost connectivity to it and the button was not longer working.

**Pull request merged**
